src/perf/lib/ThroughputClient.cpp

```diff
@@ -0,0 +1,410 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Throughput Client Implementation.
+
+--*/
+
+#ifdef QUIC_CLOG
+#include "ThroughputClient.cpp.clog.h"
+#endif
+
+#ifndef _KERNEL_MODE
+#define QUIC_TEST_APIS 1
+#endif
+#include "msquichelper.h"
+#include "ThroughputClient.h"
+#include "ThroughputCommon.h"
+
+static uint8_t* RawIoBuffer{nullptr};
+
+ThroughputClient::ThroughputClient(
+    ) {
+    QuicZeroMemory(&LocalIpAddr, sizeof(LocalIpAddr));
+    if (Session.IsValid()) {
+        Session.SetAutoCleanup();
+    }
+}
+
+QUIC_STATUS
+ThroughputClient::Init(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[]
+    ) {
+    Port = THROUGHPUT_DEFAULT_PORT;
+    TryGetValue(argc, argv, "port", &Port);
+
+    const char* Target;
+    if (!TryGetValue(argc, argv, "target", &Target)) {
+        WriteOutput("Must specify '-target' argument!\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    uint16_t ip;
+    if (TryGetValue(argc, argv, "ip", &ip)) {
+        switch (ip) {
+        case 4: RemoteFamily = AF_INET; break;
+        case 6: RemoteFamily = AF_INET6; break;
+        }
+    }
+
+    TryGetValue(argc, argv, "length", &Length);
+    if (Length == 0) {
+        WriteOutput("Must specify a positive 'length'\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    const char* localAddress;
+    if (TryGetValue(argc, argv, "bind", &localAddress)) {
+        if (!ConvertArgToAddress(localAddress, 0, &LocalIpAddr)) {
+            WriteOutput("Failed to decode IP address: '%s'!\nMust be *, a IPv4 or a IPv6 address.\n", localAddress);
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+    }
+
+    // TODO: Core, since we need to support kernel mode
+    #if _WIN32 && !_KERNEL_MODE
```

Comment:
As we discussed on our call, use QUIC_COMPARTMENT_ID for all this.

Reply:
Done.